src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -240,12 +240,37 @@ func parseQuery(r *http.Request) (string, error) {
 	return queries[0], nil
 }
 
+func sanitizeSeries(series []*ts.Series) []*ts.Series {
+	filtered := series[:0]
+	for _, s := range series {
+		dps := s.Values().Datapoints()
+		hasVal := false
+		for _, dp := range dps {
+			if !math.IsNaN(dp.Value) {
+				hasVal = true
+				break
+			}
+		}
+
+		if hasVal {
+			filtered = append(filtered, s)
+		}
+	}
+
+	return filtered
+}
+
 func renderResultsJSON(
 	w io.Writer,
 	series []*ts.Series,
 	params models.RequestParams,
 	keepNans bool,
```

Comment:
Is this ever true?  If not, can we remove the option entirely perhaps?

Reply:
I'm pretty happy to remove it since I don't really know why you'd want to keep nans, but you'll have to convert Ben :p

Reply:
Oh, if you're asking if it can theoretically be true, there's a config option to keep nans in the output; don't know why anyone would turn it on, but it's there :p